File: app/backend/src/frameworks_and_drivers/persistence/entities/product_image_entity.rs
use sqlx::FromRow;

/// 製品画像エンティティ
/// Clean Architecture: Frameworks & Drivers層のデータ表現
#[derive(Debug, FromRow)]
pub struct ProductImageEntity {
    pub id: i64,                           // INTEGER PK
    pub product_id: String,                // 外部キー (UUID)
    pub image_url: String,
    pub alt_text: Option<String>,
    pub display_order: i32,
    pub created_at: String,
    pub updated_at: String,
} 

File: app/backend-ts/src/infrastructure/database/entities/product-image.entity.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  ManyToOne,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm';

import { ProductEntity } from './product.entity';

@Entity('product_images')
export class ProductImageEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column('text')
  product_id: string;

  @Column('text')
  image_url: string;

  @Column('text', { nullable: true })
  alt_text?: string;

  @Column('integer', { default: 0 })
  display_order: number;

  @CreateDateColumn()
  created_at: Date;

  @UpdateDateColumn()
  updated_at: Date;

  @ManyToOne(() => ProductEntity, (product) => product.images)
  @JoinColumn({ name: 'product_id' })
  product: ProductEntity;
}
